sycl/plugins/unified_runtime/ur/adapters/hip/usm.cpp

```diff
@@ -207,3 +274,198 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMReleaseExp(ur_context_handle_t Context,
+                                                  uint32_t Alignment) {
+  return USMSharedAllocImpl(ResultPtr, Context, Device, nullptr, nullptr, Size,
+                            Alignment);
+}
+
+ur_result_t USMDeviceMemoryProvider::allocateImpl(void **ResultPtr, size_t Size,
+                                                  uint32_t Alignment) {
+  return USMDeviceAllocImpl(ResultPtr, Context, Device, nullptr, Size,
+                            Alignment);
+}
+
+ur_result_t USMHostMemoryProvider::allocateImpl(void **ResultPtr, size_t Size,
+                                                uint32_t Alignment) {
+  return USMHostAllocImpl(ResultPtr, Context, nullptr, Size, Alignment);
+}
+
+ur_usm_pool_handle_t_::ur_usm_pool_handle_t_(ur_context_handle_t Context,
+                                             ur_usm_pool_desc_t *PoolDesc)
+    : Context(Context) {
+  const void *pNext = PoolDesc->pNext;
+  while (pNext != nullptr) {
+    const ur_base_desc_t *BaseDesc = static_cast<const ur_base_desc_t *>(pNext);
+    switch (BaseDesc->stype) {
+    case UR_STRUCTURE_TYPE_USM_POOL_LIMITS_DESC: {
+      const ur_usm_pool_limits_desc_t *Limits =
+          reinterpret_cast<const ur_usm_pool_limits_desc_t *>(BaseDesc);
+      for (auto &config : DisjointPoolConfigs.Configs) {
+        config.MaxPoolableSize = Limits->maxPoolableSize;
+        config.SlabMinSize = Limits->minDriverAllocSize;
+      }
+      break;
+    }
+    default: {
+      throw UsmAllocationException(UR_RESULT_ERROR_INVALID_ARGUMENT);
+    }
+    }
+    pNext = BaseDesc->pNext;
+  }
+
+  auto MemProvider =
+      umf::memoryProviderMakeUnique<USMHostMemoryProvider>(Context, nullptr)
+          .second;
+
+  HostMemPool =
+      umf::poolMakeUnique<usm::DisjointPool, 1>(
+          {std::move(MemProvider)},
+          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Host])
+          .second;
+
+  auto Device = Context->DeviceId;
+  MemProvider =
+      umf::memoryProviderMakeUnique<USMDeviceMemoryProvider>(Context, Device)
+          .second;
+  DeviceMemPool =
+      umf::poolMakeUnique<usm::DisjointPool, 1>(
+          {std::move(MemProvider)},
+          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Device])
+          .second;
+
+  MemProvider =
+      umf::memoryProviderMakeUnique<USMSharedMemoryProvider>(Context, Device)
+          .second;
+  SharedMemPool =
+      umf::poolMakeUnique<usm::DisjointPool, 1>(
+          {std::move(MemProvider)},
+          this->DisjointPoolConfigs.Configs[usm::DisjointPoolMemType::Shared])
+          .second;
+  Context->addPool(this);
+}
+
+bool ur_usm_pool_handle_t_::hasUMFPool(umf_memory_pool_t *umf_pool) {
+  return DeviceMemPool.get() == umf_pool || SharedMemPool.get() == umf_pool ||
+         HostMemPool.get() == umf_pool;
+}
+
+UR_APIEXPORT ur_result_t UR_APICALL urUSMPoolCreate(
+    ur_context_handle_t Context, ///< [in] handle of the context object
```

Comment:
Not sure if the comments align to the style guide?

Reply:
No problem! I'll just leave it for now as it's the same in a few different files here and maybe a future PR could be opened to make commenting consistent throughout.